### ISSUE TITLE
Wings CRON jobs fails to send activity logs to Panel because of invalid IP field

### ISSUE THREAD

**Issue by hugotrsd** [pterodactyl/panel#4653]
### Current Behavior

CRON job on Wings node sends incorrect data to Panel, resulting in Panel rejecting activity logs.

```
ERROR: [Jan 14 12:00:04.225] activity process failed to execute cron=activity error=cron: failed to send activity events to Panel: Error response from Panel: ValidationException: The data.0.ip must be a valid IP address. (HTTP/422) subsystem=cron
```

A dump of `/var/lib/pterodactyl/wings.db` reveals invalid IPv6 addresses are being recorded in the table of pending activities.
```
sqlite> .schema
CREATE TABLE `activities` (`id` integer NOT NULL,`user` uuid,`server` uuid NOT NULL,`event` text NOT NULL,`metadata` text,`ip` text NOT NULL,`timestamp` datetime NOT NULL,PRIMARY KEY (`id`));
CREATE INDEX `idx_activities_event` ON `activities`(`event`);
sqlite> select * from activities;
1|6bfc3eaa-25dd-489f-bfdb-8a89d18a9a4b|26f8829c-ee8a-4cb1-bcb8-0c80c42884c7|server:console.command|{"command":"help"}|fda1:ebc0:b0ce:7310:|2023-01-14 10:59:42.885410019+00:00
```
where `fda1:ebc0:b0ce:7310:` is not a valid IPv6 address and should be `fda1:ebc0:b0ce:7310::3` in this context.

Updating the faulty record with `UPDATE activities SET ip = 'fda1:ebc0:b0ce:7310::3'` solves the issue until another activity gets recorded. If the client triggering the activity log uses IPv4, the issue does not occur.

While similar to #4290, this issue seems to rather be caused by how data gets recorded to the database.

### Expected Behavior

No exception. Activity logs are accepted by Panel.

### Steps to Reproduce

Generate an activity log (stop/start a server, send a command to a server, ..) while using IPv6 to communicate with the Wings node.

### Panel Version

1.11.2

### Wings Version

1.11.0

### Games and/or Eggs Affected

Tested with Minecraft (Vanilla) only, but probably affects all

### Docker Image

ghcr.io/pterodactyl/yolks:java_17

### Error Logs

```bash
Wings: https://ptero.co/disufoleqy
Panel: Unavailable

I'm connecting to Wings and Panel though my own reverse proxy (`fda1:ebc0:b0ce:7310::3`), but Wings and Panel are able to talk to each other directly. All three of them are running in LXC.
```

### Is there an existing issue for this?

- [X] I have searched the existing issues before opening this issue.
- [X] I have provided all relevant details, including the specific game and Docker images I am using if this issue is related to running a server.
- [X] I have checked in the Discord server and believe this is a bug with the software, and not a configuration issue with my specific system.

**Comment by hwalker928** [comment]
Not sure if this is related or even accurate, but it seems LXC is not supported by Wings: https://pterodactyl.io/wings/1.0/installing.html#system-requirements

**Comment by devnote-dev** [comment]
This is the same as #4290, the `ip` and `user` fields are nullable (you can reproduce the user one with SFTP activity logs IIRC), however, the endpoint for remote activities in the panel has incorrect validation rules.

**Comment by hugotrsd** [comment]
I believe it really isn't the same, since the IP field is not null in this case. The validation rules are actually doing their job here, rightfully informing Wings that the value for `ip` is not a valid IP address.

The `ip` field is actually marked as `NOT NULL` in the database, as you can see from the dump. I think the issue is related to how Wings handles and writes IPv6 addresses to the database, rather than how the Panel processes the data sent by Wings.

**Comment by devnote-dev** [comment]
Oh I see what you mean, the IP looks valid to me though?: https://laravalidation.com/x4OP0n

**Comment by hugotrsd** [comment]
The value in the database is `fda1:ebc0:b0ce:7310:`, which is invalid.
I replaced it with `fda1:ebc0:b0ce:7310::3` using the SQLite CLI to check if that would solve the issue, and it did.

**Comment by devnote-dev** [comment]
Right, my bad. The cronjob for activities are handled here ([activity_cron.go#L33-L36](https://github.com/pterodactyl/wings/blob/develop/internal/cron/activity_cron.go#L33-L36)), can you run that query in SQLite and see what it returns?

**Comment by hugotrsd** [comment]
Here's what I got:
```
sqlite> .schema
CREATE TABLE `activities` (`id` integer NOT NULL,`user` uuid,`server` uuid NOT NULL,`event` text NOT NULL,`metadata` text,`ip` text NOT NULL,`timestamp` datetime NOT NULL,PRIMARY KEY (`id`));
CREATE INDEX `idx_activities_event` ON `activities`(`event`);
sqlite> select * from activities;
1|6bfc3eaa-25dd-489f-bfdb-8a89d18a9a4b|26f8829c-ee8a-4cb1-bcb8-0c80c42884c7|server:console.command|{"command":"help"}|fda1:ebc0:b0ce:7310:|2023-01-14 10:59:42.885410019+00:00
2|6bfc3eaa-25dd-489f-bfdb-8a89d18a9a4b|26f8829c-ee8a-4cb1-bcb8-0c80c42884c7|server:console.command|{"command":"help"}|fd02:842a:8b:4601::f8f8|2023-01-14 15:24:51.065161855+00:00
sqlite> select * from activities where event not like 'server:sftp.%';
1|6bfc3eaa-25dd-489f-bfdb-8a89d18a9a4b|26f8829c-ee8a-4cb1-bcb8-0c80c42884c7|server:console.command|{"command":"help"}|fda1:ebc0:b0ce:7310:|2023-01-14 10:59:42.885410019+00:00
2|6bfc3eaa-25dd-489f-bfdb-8a89d18a9a4b|26f8829c-ee8a-4cb1-bcb8-0c80c42884c7|server:console.command|{"command":"help"}|fd02:842a:8b:4601::f8f8|2023-01-14 15:24:51.065161855+00:00
```

You will notice a second record with index 2, which actually has a valid IPv6 address in it: this is me debugging.
It seems like when I add my proxy to the list of `trusted_proxies` in the `config.yml` file of Wings, the `ip` field gets appropriately populated with the value from the `Forwarded:` HTTP header. Weird.

**Comment by danny6167** [comment]
Interesting. I have a couple of ideas, but will have to wait until I can spend some time screwing around with GO tomorrow night.
I think there is an upstream bug that is not handling IPv6 addresses correctly and it's trimming off what it *thinks* is a port number from the address.

**Comment by hugotrsd** [comment]
I spent my day investigating this exact hypothesis looking at `gin` code, but I think I don't know enough Go to follow this through unfortunately... Hope your eyes will be sharper than mine!

Here's what I got, if that can help:
- Wings doesn't seem to do very much with the client IP address from what I've figured. It just grabs it from `gin` and passes it to `gorm` for database storage.
- If the issue was upstream, I would expected `gin.ClientIP()` to return something invalid. But from what I saw, `gin` only makes calls to standard packages like `net` and `string` to process IP addresses, and I wouldn't expect those to be the root cause...
- `net.SplitHostPort()` seems to correctly throw an error on missing `[backets]` in the IPv6 string.
- [This call](https://github.com/gin-gonic/gin/blob/3010cbd7f4eccdbb610c510274895e083b8c058c/context.go#L791) to `net.ParseIP()` should prevent any ill-formed IP to get though, no?
- I enabled debug logging in Wings -> [this call](https://github.com/pterodactyl/wings/blob/e74d8e35014cd545802d04e50541921898113abb/router/router.go#L32-L37) logs the correct IPv6 address.
